### ISSUE TITLE
[no-use-before-define] false positive when exported var name corresponds to interface property name

### ISSUE THREAD

**Issue by zuzusik** [nzakas/eslint-plugin-typescript#141]
```js
interface ITest {
  first : boolean;
  second : string;
  third : boolean;
}

let first = () => console.log('first');

export let second = () => console.log('second');

export namespace Third {
  export let third = () => console.log('third');
}
```

leads to 

```
  2:3  error  'first' was used before it was defined   typescript/no-use-before-define
  3:3  error  'second' was used before it was defined  typescript/no-use-before-define
  4:3  error  'third' was used before it was defined   typescript/no-use-before-define
```

Note: removing both exports leads to error for non exported member be gone, so:
```js
interface ITest {
  first : boolean;
  second : string;
  third : boolean;
}

let first = () => console.log('first');
```
passes just fine
